de.tudarmstadt.ukp.wikipedia.api/src/main/java/de/tudarmstadt/ukp/wikipedia/api/hibernate/GenericDAO.java

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright 2017
+ * Ubiquitous Knowledge Processing (UKP) Lab
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package de.tudarmstadt.ukp.wikipedia.api.hibernate;
+
+import de.tudarmstadt.ukp.wikipedia.api.Wikipedia;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hibernate.LockOptions;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+
+/**
+ * A common base class for DAO classes.
+ * 
+ * @param <T> The entity type to provide persistence features for.
+ */
+public abstract class GenericDAO<T> {
+
+    private final Log logger = LogFactory.getLog(getClass());
+    
+    private final Wikipedia wiki;
+    private final SessionFactory sessionFactory;
+
+    private final String entityClass;
+
+    GenericDAO(Wikipedia wiki, String entityClass) {
+        this.wiki = wiki;
+        this.entityClass = entityClass;
+        this. sessionFactory = initializeSessionFactory();
+    }
+
+    private SessionFactory initializeSessionFactory() {
+        try {
+            return WikiHibernateUtil.getSessionFactory(wiki.getDatabaseConfiguration());
+        } catch (Exception e) {
+            logger.error("Could not locate SessionFactory in JNDI", e);
+            throw new IllegalStateException("Could not locate SessionFactory in JNDI");
+        }
+    }
+
+    private SessionFactory getSessionFactory() {
+        return sessionFactory;
+    }
+
+    protected Session getSession() {
+        return getSessionFactory().getCurrentSession();
+    }
+
+    public void persist(T transientInstance) {
+        try {
+            getSession().persist(transientInstance);
+            logger.trace("persist successful");
+        } catch (RuntimeException re) {
+            logger.error("persist failed", re);
```

Comment:
Why log if the exception is thrown up anyway? (general comment).

Reply:
Again, existing code. I will remove the logger in the `catch` block for all occurrences in this class and push this and other improvements with the next commit.

Reply:
My last comment is now somewhat blocked by the comment below (on optimizing logger output) for merge failures. How to proceed?

Reply:
In general, if a problem was identified and acknowledged and is for some reason deemed outside of the PR, then record it as an issue and set up a new PR, either based off the original PR branch or off the master once the PR has been merged.